include/revng/Support/DataBuffer.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "llvm/ADT/SmallString.h"
+#include "llvm/Support/MemoryBuffer.h"
+
+// Souped-up llvm::ArrayRef which overloads for both uint8_t and char "array"
+// datastructures
+class DataBuffer {
```

Comment:
> I'd be happier if we didn't have this.

I'd be too, but it's really annoying because half the classes use `char *` and the other half use `uint8_t`, so it's either having `reinterpret_cast` in a bunch of places or all concentrated in this file

Reply:
Dropped the class, implemented via a concept